Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -17,15 +17,60 @@ import ArcGIS
 
 /// An object that encapsulates state about a preplanned map.
 @MainActor
-class PreplannedMapModel: ObservableObject, Identifiable {
+public class PreplannedMapModel: ObservableObject, Identifiable {
     /// The preplanned map area.
     let preplannedMapArea: any PreplannedMapAreaProtocol
     
+    /// The task to use to take the area offline.
+    private let offlineMapTask: OfflineMapTask
+    
+    /// The download directory for the preplanned map areas.
+    private let preplannedDirectory: URL
+    
+    /// The ID of the preplanned map area.
+    private let  preplannedMapAreaID: String
+    
+    /// The mobile map package for the preplanned map area.
+    private(set) var mobileMapPackage: MobileMapPackage?
+    
+    /// The currently running download job.
+    @Published private(set) var job: DownloadPreplannedOfflineMapJob?
+    
     /// The combined status of the preplanned map area.
     @Published private(set) var status: Status = .notLoaded
     
-    init(preplannedMapArea: PreplannedMapAreaProtocol) {
-        self.preplannedMapArea = preplannedMapArea
+    /// The result of the download job. When the result is `.success` the mobile map package is returned.
+    /// If the result is `.failure` then the error is returned. The result will be `nil` when the preplanned
+    /// map area is still packaging or loading.
+    @Published private(set) var result: Result<MobileMapPackage, Error>?
+    
+    /// A Boolean value indicating if download can be called.
+    var canDownload: Bool {
+        switch status {
+        case .notLoaded, .loading, .loadFailure, .packaging, .packageFailure,
+                .downloading, .downloaded:
+            false
+        case .packaged, .downloadFailure:
+            true
+        }
+    }
+    
+    init?(
+        offlineMapTask: OfflineMapTask,
+        mapArea: PreplannedMapAreaProtocol,
+        directory: URL
+    ) {
+        self.offlineMapTask = offlineMapTask
+        preplannedMapArea = mapArea
+        preplannedDirectory = directory
+        
+        if let itemID = preplannedMapArea.id {
+            preplannedMapAreaID = itemID.rawValue
+        } else {
+            return nil
+        }
+        
+        setDownloadJob()
```

Comment:
This looks like a side effect that shouldn't be in the initializer. I'll tinker with it in my local branch and see what I can change.

Reply:
The initializer should, at the very least, be doc'd why it would return `nil`. However, a potentially better way to do this is to take in a non-optional preplanned map area id, then the init does not need to be bailable.